app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/cloud/apis/AppEngineCloudApiActionDecorator.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.java.cloud.apis;
+
+import com.google.cloud.tools.intellij.appengine.java.AppEngineMessageBundle;
+import com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacet;
+import com.google.cloud.tools.intellij.appengine.java.facet.standard.AppEngineStandardFacetType;
+import com.google.cloud.tools.intellij.appengine.java.project.AppEngineProjectService;
+import com.google.cloud.tools.intellij.cloudapis.CloudApiActionDecoratorExtension;
+import com.google.cloud.tools.intellij.cloudapis.GoogleCloudApisMessageBundle;
+import com.intellij.facet.FacetManager;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.JavaVersion;
+
+/**
+ * {@link CloudApiActionDecoratorExtension} extension point that checks for Java projects version if
+ * App Engine is used. Otherwise, action is disabled and help text is shown.
+ */
+public class AppEngineCloudApiActionDecorator implements CloudApiActionDecoratorExtension {
+
+  @Override
+  public boolean decorate(AnActionEvent e) {
```

Comment:
can you add a javadoc on this one perhaps? specifically I think it would be nice to see an explanation on what the boolean return value represents for this specific decorator implementation

Reply:
done.